skmediate/conditional_independence.py

```diff
@@ -42,37 +127,76 @@ def __init__(
                301-310, DOI: 10.1080/19466315.2019.1575276
         """
         if regression_estimator is None:
-            self.regression_estimator_xz = (
-                self.regression_estimator_xy
-            ) = LinearRegression()
+            self.regression_estimator_xz = LinearRegression()
+            self.regression_estimator_xy = LinearRegression()
         elif isinstance(regression_estimator, collections.Sequence):
+            if not all(isinstance(r, RegressorMixin) for r in regression_estimator):
+                mro = [getmro(type(r)) for r in regression_estimator]
+                raise ValueError(
+                    f"regression_estimator must contain estimator instances that "
+                    f"inherit from sklearn.base.RegressorMixin. Got "
+                    f"{regression_estimator} instead. These instances have the "
+                    f"following method resolution order:\n{mro}"
+                )
             self.regression_estimator_xz = regression_estimator[0]
             self.regression_estimator_xy = regression_estimator[1]
         else:
-            self.regression_estimator_xz = (
-                self.regression_estimator_xy
-            ) = regression_estimator
+            if not isinstance(regression_estimator, RegressorMixin):
```

Comment:
Maybe we can loosen this up a bit, or add an argument to ignore this error? We really want anything with a fit and predict method. There are multiple useful tools that look like scikit learn models but do not necessarily inherit the RegressorMixin.

Reply:
True. We could relax this. One caveat is that we use `sklearn.base.clone` to make sure that the xy and xz regressors don't get mixed up. If we want to keep this, then we have two options:

1. Use `safe=False` in the parameters to `clone`. This will reproduce the current behavior for sklearn estimators and then fall back to a deep copy on objects that are not estimators. Not sure if this have the intended effect.
2. In addition to requiring a `fit` and `predict` method, we also require a `get_params` method so that it works with `clone` without setting `safe=False`. This would work but would place more burden on the user. Personally I would inherit from `RegressorMixin` rather than write my own `get_params` method.

Option 1 is the easiest for us to implement and probably gives the most flexibility to the user.

@arokem, @36000: thoughts?

Reply:
My thinking is that we would probably gain some clarity from enforcing full compliance with the `RegressorMixin`. Arguably, if a user (even one of us) wants to pass in something that does not inherit from `RegressorMixin`, they should first wrap that thing in an object that does inherit from `RegressorMixin`, make sure that works as expected, and then proceed from there. So, if I had to choose between these two options, I would probably go with 2. Is that too strict?

Reply:
I went with 2 in the latest commit.